Use property settings to attach mouse and touch events.

youngca opened the pull request:
For each kinetic object class, add properties of supportMouseEvents and supportTouchEvents.
If these properties are true, then when the object is created and initialized, the mouse and/or touch events will be added.

@scottnc27603 @davideaster

youngca committed:
Use property settings to attach mouse and touch events.

For each kinetic object class, add properties of supportMouseEvents and supportTouchEvents.
If these properties are true, then when the object is created and initialized, the mouse and/or touch events will be added.

scottnc27603 commented:
We need to protect against having the events added more that one time.  Maybe storing a value for each type in the shared node object would be a potential solution.  Those flags could be added in createLocalNode in the model/kinetic.js and then all of the objects will get those values.

@youngca

youngca committed:
Double check to ensure the mouse and touch events are only added once per object.

youngca committed:
Rename the return variable to something more obvious.

youngca committed:
Propagate the status of attached mouse and touch events in kinetic nodes.

Add a property to the kinetic node, set it to false in the model and update it in the view when the events are attached.

youngca commented:
Status of whether the mouse and/or touch events have been attached are stored in the shared node.

@scottnc27603 @davideaster

scottnc27603 commented:
:+1:

youngca committed:
Mouse and touch event addition only when needed per class.

Turn off listening by default - object will not listen to mouse and touch events until listening is turned on.
Register when mouse and touch events have been added to the object.

Also, mil-sym symbols support roles in creation requests.
